lib/pure/strutils.nim

```diff
@@ -209,7 +209,7 @@ proc toLowerAscii*(c: char): char {.noSideEffect,
     doAssert toLowerAscii('A') == 'a'
     doAssert toLowerAscii('e') == 'e'
   if c in {'A'..'Z'}:
-    result = chr(ord(c) + (ord('a') - ord('A')))
+    result = cast[char](cast[uint8](c) xor 0b0010_0000'u8)
```

Comment:
```suggestion
    result = char(uint8(c) xor 0b0010_0000'u8)
```

Reply:
why do convert ? casting is not just fine there?

Reply:
It just looks ugly, and should really only be used for unsafe memory operations afaik

Reply:
I prefer cast, when you do convert in other types they do different operation produce different results. cast match my intention, I dont meant to convert one data type to other data type.